library/std/src/sync/once.rs

```diff
@@ -395,12 +395,13 @@ impl Once {
                 }
                 POISONED | INCOMPLETE => {
                     // Try to register this thread as the one RUNNING.
-                    let old = self.state_and_queue.compare_and_swap(
+                    let exchange_result = self.state_and_queue.compare_exchange(
```

Comment:
could this use `_weak`?

Reply:
I think it might be OK to use `_weak` here. I changed it into that. But I hope more people can step in and verify this before it's merged.